anta/tests/routing/bgp.py

```diff
@@ -1226,3 +1226,81 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"Following BGP peers are not configured or hold and keep-alive timers are not correct:\n{failures}")
+
+
+class VerifyBGPPeerErrors(AntaTest):
+    """Verifies if the BGP peer's errors counters should be zero.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the BGP peer's error counters are zero.
+    * Failure: The test will fail if the BGP peer's error counters are non-zero.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.routing:
+      bgp:
+        - VerifyBGPPeerErrors:
+            bgp_peers:
+              - peer_address: 172.30.11.1
+                vrf: default
+                errors:
+                  - inUpdErrWithdraw
+    ```
+    """
+
+    name = "VerifyBGPPeerErrors"
+    description = "Verifies the error counters of a BGP peer."
+    categories: ClassVar[list[str]] = ["bgp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="show bgp neighbors {peer} vrf {vrf}")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyBGPPeerErrors test."""
+
+        bgp_peers: list[BgpPeer]
+        """List of BGP peers"""
+
+        class BgpPeer(BaseModel):
+            """Model for a BGP peer."""
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer."""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            errors: list[BgpErrors]
```

Comment:
Like #778, we should have something like:
```
update_error_filter: list[BgpUpdateError] | None = None
```
None means we check all the errors, other we check only the provided ones.

Reply:
Added check for verifying all the error counters and if any mentioned error counter not found.